excel/src/main/java/org/apache/metamodel/excel/ExcelUtils.java

```diff
@@ -237,13 +292,34 @@ public static String getCellValue(Workbook wb, Cell cell) {
         return result;
     }
 
+    public static Object getCellValueChecked(final Workbook wb, final Cell cell, final ColumnType expectedColumnType) {
+        final Object value = getCellValueAsObject(wb, cell);
+        if (value == null || value.getClass().equals(expectedColumnType.getJavaEquivalentClass())) {
+            return value;
+        }
+
+        if (!(value.getClass().equals(Integer.class) && expectedColumnType
+                .getJavaEquivalentClass()
+                .equals(Double.class)) && logger.isWarnEnabled()) {
+            // Don't log when a Integer value is in a Double column type
```

Comment:
Can you move this outside the if, it applies to the if, and in this manner, it's inside the scope of the resolved if statement. Also can you change "a Integer" to "an Integer"?

Reply:
done